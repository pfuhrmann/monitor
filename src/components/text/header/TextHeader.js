import React from 'react';
import PropTypes from 'prop-types';
import './TextHeader.scss';

const TextHeader = ({ children, className, ...otherProps }) => (
  <h2 className={className} {...otherProps}>{children}</h2>
);

TextHeader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

TextHeader.defaultProps = {
  children: '',
  className: '',
};

export default TextHeader;
