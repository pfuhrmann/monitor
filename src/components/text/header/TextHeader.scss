@import "../../../variables.scss";

h1 {
  font-size: 2rem;
  font-weight: $font-weight-regular;
}
