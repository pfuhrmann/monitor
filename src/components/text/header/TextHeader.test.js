import React from 'react';
import ReactDOM from 'react-dom';
import TextHeader from './TextHeader';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextHeader />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly h2 tag with text', () => {
  const component = renderer.create(
    <TextHeader>Header</TextHeader>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
