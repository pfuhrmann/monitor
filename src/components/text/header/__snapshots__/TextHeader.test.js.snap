// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly h2 tag with text 1`] = `
<h2
  className=""
>
  Header
</h2>
`;
